libs/seiscomp/math/filter/iirintegrate.cpp

```diff
@@ -71,9 +71,13 @@ void IIRIntegrate<T>::setSamplingFrequency(double fsamp) {
 
 	double dt = 1./fsamp;
 
-	_a0 = _ia0 * dt;
+  double a0 = _ia0 * dt;
```

Comment:
Does this block intent to check if the sampling frequency has changed? Why not just checking it at the top of the function as with all other filters?

Reply:
Yes. I didn't like it, neither. Instead, it is much cleaner to store the sampling frequency as a member variable and do the check as with all other filters.

Reply:
OK, I see. Adding _fsamp will break binary compability. So this might be the best approach.

Reply:
I went for the original attempt with a minor change.

>  Adding _fsamp will break binary compability.

Again, I learned something :) ...